controllers/apps/operations/reconfigure_pipeline.go

```diff
@@ -111,7 +114,15 @@ func (p *pipeline) ConfigConstraints() *pipeline {
 			Name: p.configSpec.ConfigConstraintRef,
 		}
 		p.configConstraint = &appsv1beta1.ConfigConstraint{}
-		return p.cli.Get(p.reqCtx.Ctx, ccKey, p.configConstraint)
+		if err := p.cli.Get(p.reqCtx.Ctx, ccKey, p.configConstraint); err != nil {
+			return err
+		}
+		if p.configConstraint.Spec.FileFormatConfig == nil {
```

Comment:
since configConstraint.Spec.FileFormatConfig is required, why its value can become nil?

Reply:
This error generally does not occur and may be related to the upgrade. I plan to fix this because it is not appropriate for kubeblock to panic directly.

Reply:
I see, it's a validation check. Is there a more appropriate place to perform these checks?